bin/test

```diff
@@ -1,9 +1,31 @@
 #!/usr/bin/env bash
 
 BUILD_DIR=${1:-}
-if pnpm --version > /dev/null 2>&1; then
+
+read_package_json() {
+  local key="$1"
+  # shellcheck disable=SC2002
+  cat "$BUILD_DIR/package.json" | jq -c -M --raw-output "$key // \"\"" || return 1
+}
+
+which_tool() {
+  yarn_engine=$(read_package_json ".engines.yarn")
+  pnpm_engine=$(read_package_json ".engines.pnpm")
+  package_manager=$(read_package_json ".packageManager")
+  if [ -n "$pnpm_engine" ] || [[ "$package_manager" == pnpm* ]]; then
+    echo "pnpm"
+  elif [ -n "$yarn_engine" ] || [[ "$package_manager" == yarn* ]]; then
+    echo "yarn"
+  else
```

Comment:
Should we handle conflicts in some way? What if both `engines.yarn` and `engines.pnpm` are set? What if `engines.yarn` is set, but `packageManager` is `pnpm@*`?

Reply:
Good question. I was thinking that case would result in an error earlier in the `compile` (or `test-compile`) process but we only check and fail if there are multiple lockfiles, not conflicting package manager metadata. I'll add some coverage for this possibility too.

Reply:
The above should be addressed by #1317